generator/src/test/java/com/scottlogic/deg/generator/reducer/ConstraintReducerTest.java

```diff
@@ -1001,6 +1001,26 @@ void shouldReduceMatchesRegexWithSingletonConstraint() {
                 Matchers.greaterThan(0L));
     }
 
+    @Test
+    void shouldReduceToConstraintsFoundInDecisions() {
+        final Field field = new Field("test0");
+
+        List<AtomicConstraint> decisionConstraints = Arrays.asList(
+            new IsEqualToConstantConstraint(field, "a", rules()));
+
+        Optional<FieldSpec> testOutput = constraintReducer.reduceConstraintsToFieldSpec(
+            Collections.emptySet(),
+            decisionConstraints
+        );
+
+        FieldSpec outputSpec = testOutput.get();
+
+        Assert.assertThat("Fieldspec is not empty", outputSpec,
+            not(equalTo(FieldSpec.Empty)));
+        Assert.assertThat("Fieldspec has must contain restriction",
+            outputSpec.getMustContainRestriction(), not(nullValue()));
```

Comment:
Is it possible to make this assertion more explicit, not(nullValue()) isn't ideal

Reply:
Resolved, more indepth assertions added